postfix.grok

```diff
@@ -16,6 +16,7 @@ POSTFIX_TLSCONN (Anonymous|Trusted|Untrusted|Verified) TLS connection establishe
 POSTFIX_DELAYS %{NUMBER:postfix_delay_before_qmgr}/%{NUMBER:postfix_delay_in_qmgr}/%{NUMBER:postfix_delay_conn_setup}/%{NUMBER:postfix_delay_transmission}
 POSTFIX_LOSTCONN (lost connection|timeout|Connection timed out)
 POSTFIX_PROXY_MESSAGE (%{POSTFIX_STATUS_CODE:postfix_proxy_status_code} )?(%{POSTFIX_STATUS_CODE_ENHANCED:postfix_proxy_status_code_enhanced})?.*
+POSTFIX_CONNSTATUS %{POSTFIX_QUEUEID:postfix_queueid}: host %{POSTFIX_RELAY_INFO} said: %{POSTFIX_STATUS_CODE:postfix_status_code} %{POSTFIX_STATUS_CODE_ENHANCED:postfix_status_code_enhanced} <%{GREEDYDATA:postfix_sender_address}>: %{GREEDYDATA:postfix_connect_status}
```

Comment:
Hi Tom,

thanks for explanation. Maybe it would be possible to keep rules in sync
with major MTAs.
Anyway, one may add custom rules right after 50-filter-postfix.conf and
don't mess up postfix.grok

--
С уважением,
Илья Слепнёв

2015-10-07 19:53 GMT+03:00 Tom Hendrikx <notifications@github.com>:

> In postfix.grok
> <https://github.com/whyscream/postfix-grok-patterns/pull/64#discussion_r41415474>
> :
>
> > @@ -16,6 +16,7 @@ POSTFIX_TLSCONN (Anonymous|Trusted|Untrusted|Verified) TLS connection establishe
> >  POSTFIX_DELAYS %{NUMBER:postfix_delay_before_qmgr}/%{NUMBER:postfix_delay_in_qmgr}/%{NUMBER:postfix_delay_conn_setup}/%{NUMBER:postfix_delay_transmission}
> >  POSTFIX_LOSTCONN (lost connection|timeout|Connection timed out)
> >  POSTFIX_PROXY_MESSAGE (%{POSTFIX_STATUS_CODE:postfix_proxy_status_code} )?(%{POSTFIX_STATUS_CODE_ENHANCED:postfix_proxy_status_code_enhanced})?.*
> > +POSTFIX_CONNSTATUS %{POSTFIX_QUEUEID:postfix_queueid}: host %{POSTFIX_RELAY_INFO} said: %{POSTFIX_STATUS_CODE:postfix_status_code} %{POSTFIX_STATUS_CODE_ENHANCED:postfix_status_code_enhanced} <%{GREEDYDATA:postfix_sender_address}>: %{GREEDYDATA:postfix_connect_status}
>
> The problem with relay response parsing is that you'll never know if the
> relay is postfix, or sendmail, or any other MTA, which means you cannot
> even depend on a common format in status_code, enhanced_status_code and
> the email address (postfix_sender_address?), as your PR assumes. It works
> for some upstreams (i.e. where the upstream is of postfix), but since the
> upstream MTA is not part of the system you're monitoring (the local MTA), I
> decided to only extract the complete response in a single field, and don't
> try to parse it any further.
>
> Some work on this was done at bfb4cc1
> <https://github.com/whyscream/postfix-grok-patterns/commit/bfb4cc19421916392e4d59d2e5f0639c308cf0f8>,
> but that is still incomplete. I'll try to give it another look later this
> week.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/whyscream/postfix-grok-patterns/pull/64/files#r41415474>
> .
>